pkg/apis/admission/types.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/kubernetes/pkg/apis/authentication"
+)
+
+// AdmissionReview describes an admission request.
+type AdmissionReview struct {
+	metav1.TypeMeta
+
+	// Spec describes the attributes for the admission request.  (Note: Since this admission controller is non-mutating
+	// the webhook should avoid setting this in its response to avoid the cost of deserializing it.)
```

Comment:
One slight tweak, instead of "should" use "may".  You can probably drop the parethensis and just do a second sentence.  

> Since admission controllers are primarily responsible for making decisions in the critical path of a request, implementors may omit the spec field from their response to minimize time spent serializing requests.  Callers must not rely on the spec field being preserved in a response.